### PR TITLE
Filter tracked time on subject type [PRO-3411]

### PR DESCRIPTION
**Pull request by adognameddog**
With an array that may contain one or more elements. NULL means no subject type.

### CODE REVIEW

**Review by tiagobastos — APPROVED** [review]

File: apiary.apib [hunk 1]
```diff
@@ -3701,6 +3701,14 @@ Get a list of tracked time.
                         + milestone
                         + ticket
                 + id: `2659dc4d-444b-4ced-b51c-b87591f604d7` (string, required)
+            + subject_types (array, optional) - Include all tracked time with one of the provided subject types. For tracked time without a subject type, provide `null`
+                + Members
+                    + company
+                    + contact
+                    + event
+                    + todo
```
Comment (mrtus) [inline apiary.apib:3709]: Q: Isn't `todo` a deprecated term? Shouldn't this be `task`?

Comment (adognameddog) [reply apiary.apib:3709]: In our API, it's still `todo`. We should change that to task, but that's very much a breaking change, witch requires a new version, and is out of scope for the moment.

**Review by mikeverf — APPROVED** [review]

File: apiary.apib (hunk 1 above)
Comment (joaoinacio) [reply apiary.apib:3709]: 🤔 I seem to recall some endpoints (as well as sideloader?) moving to `task` whenever possible while still supporting legacy `todo` when required..
Will try to find some examples.

**Edit**:
SubjectType is not one of them, so `todo` still seems pretty much required...

**Review by joaoinacio — APPROVED** [review]

**Review by ketlane — APPROVED** [review]

**Review by ArnaudWeyts — APPROVED** [review]

**Comment by ArnaudWeyts** [comment]
✅ MC'd

**Pull request merged**